File: carbon/src/main/java/carbon/widget/ArraySearchAdapter.java
package carbon.widget;

public class ArraySearchAdapter<Type> extends SearchAdapter<Type> {
    private Type[] data;

    public ArraySearchAdapter(Type[] data) {
        this.data = data;
    }

    @Override
    public int getItemCount() {
        return data.length;
    }

    @Override
    public Type getItem(int i) {
        return data[i];
    }

    @Override
    public String[] getItemWords(Type item) {
        return new String[]{item.toString()};
    }
}

File: carbon/src/main/java/carbon/widget/ArraySearchAdapter.kt
package carbon.widget

open class ArraySearchAdapter<Type> : SearchAdapter<Type> {

    private val data: Array<Type>

    constructor(data: Array<Type>) : super() {
        this.data = data
    }

    override fun getItemCount(): Int {
        return data.size
    }

    override fun getItem(i: Int): Type {
        return data[i]
    }

    override fun getItemWords(item: Type): Array<String> {
        return arrayOf(item.toString())
    }
}
